docs/source/parallelism.rst

```diff
@@ -42,6 +42,32 @@ bandwidth* of the machine.  You can measure how the achieved memory
 bandwidth changes depending on the number of processes used on your
 machine using STREAMS_.
 
+Parallel garbage collection
+===========================
+
+Firedrake objects often contain PETSc objects, but are managed by
+the Python garbage collector. It is possible that when executing in
+parallel, code will deadlock as the Python garbage collector is not
+collective over the MPI communicator that the PETSc objects are
+collective over. If you find parallel code hanging for inexplicable
+reasons, it is possible to turn off the Python garbage collector by
+including these lines in your code:
+
+.. code-block:: python
+
+    import gc
+    gc.disable()
+
+The garbage collector can be turned back on with this line:
+
+.. code-block:: python
+
+    gc.enable()
+
+More information can be found in
+`this <https://github.com/firedrakeproject/firedrake/issues/1617>`_
+issue.
+
```

Comment:
This is basically good. Can we add a warning (using `.. warning::`) that disabling the garbage collector may cause memory leaks?